### ISSUE TITLE
prefer-screen-queries: Allow use with Storybook

### ISSUE THREAD

**Issue by IanVS** [testing-library/eslint-plugin-testing-library#1124]
### What rule do you want to change?

prefer-screen-queries

### Does this change cause the rule to produce more or fewer warnings?

Fewer warnings

### How will the change be implemented?

I would like to avoid warnings from this rule when using `canvas` from Storybook interactive tests.  

Currently, the example code below triggers this lint error.  I'd like to avoid that error, while still triggering errors if `canvas` is being destructured.  Essentially, I'd like to be able to set a setting on the lint rule to look for `canvas` instead of only `screen`.

### Example code

```ts
export const Default = {
  play: async ({ canvas }) => {
    const links = canvas.getAllByRole("link"); // Avoid destructuring queries from `render` result, use `screen.getAllByRole`
  }
}
```

### How does the current rule affect the code?

Currently it triggers an error.

### How will the new rule affect the code?

It will not trigger an error in this case, but will still error for:

### Anything else?

```ts
export const Default = {
  play: async ({ canvas }) => {
    const { getAllByRole } = canvas; // Avoid destructuring queries from `render` result, use `screen.getAllByRole`
    const links = getAllByRole("link"); 
  }
}
```

### Do you want to submit a pull request to change the rule?

Yes, but need help

**Comment by Belco90** [comment]
Hey @IanVS! I don't think the mentioned rule should support this scenario.

I'm not sure in which files you are using this `canvas` argument, but `prefer-screen-queries` rule should be turned off for them in favor of using the canvas if you are always accessing the elements through such param.

**Comment by IanVS** [comment]
This is in storybook files, which use testing-library.  I understand that it's an odd scenario, I just know a lot of people do their testing in storybook tests, so I wanted to at least ask.  Thanks for the response!

**Comment by Belco90** [comment]
It's fine to colocate the tests within storybook files, but then the rules from this plugin should be tweaked, so `prefer-screen-queries` is disabled for such files, for example.

**Comment by IanVS** [comment]
Yep, understood.  It would just be nice to be able to prevent pulling matchers off of the canvas (which is the result of a testing-library `render()`), which is what this rule does, so I thought maybe it could be adjusted slightly.

I'm actually seeing that the rule only triggers _sometimes_, so I'd need to do a bit more investigating anyway.  For now I'll disable as you recommend, and look for some other way to enforce the intent of this rule.
